python/ray/autoscaler/_private/subprocess_output_util.py

```diff
@@ -345,6 +345,12 @@ def run_cmd_redirected(
             stderr_file=sys.stderr,
             use_login_shells=use_login_shells,
         )
+
+        # Do our best to flush output to terminal.
+        # See https://github.com/ray-project/ray/pull/19473.
+        sys.stdout.flush()
```

Comment:
I think this code is unreachable.

Reply:
I got that impression too, but might as well flush, just in case?

Reply:
It's not unreachable because of some branch -- it's literally immediately after a `return` statement. It will never run.

Reply:
Oh, jeez.
Let's put it in a try-finally.

Reply:
Fixed.